internal/cri/server/images/image_pull.go

```diff
@@ -334,11 +334,30 @@ func (c *CRIImageService) createImageReference(ctx context.Context, name string,
 	} else if !errdefs.IsAlreadyExists(err) {
 		return err
 	}
-	if oldImg.Target.Digest == img.Target.Digest && oldImg.Labels[crilabels.ImageLabelKey] == labels[crilabels.ImageLabelKey] {
+	// Retrieve oldImg from image store here because Create routine returns an
+	// empty image on ErrAlreadyExists
+	oldImg, err := c.images.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+	var fieldpaths []string
+	if oldImg.Target.Digest != img.Target.Digest {
+		fieldpaths = append(fieldpaths, "target")
+	}
+	if oldImg.Labels[crilabels.ImageLabelKey] != labels[crilabels.ImageLabelKey] {
+		fieldpaths = append(fieldpaths, "labels."+crilabels.ImageLabelKey)
+	}
+	if oldImg.Labels[crilabels.PinnedImageLabelKey] != labels[crilabels.PinnedImageLabelKey] &&
+		labels[crilabels.PinnedImageLabelKey] == crilabels.PinnedImageLabelValue {
+		fieldpaths = append(fieldpaths, "labels."+crilabels.PinnedImageLabelKey)
+	}
+	if len(fieldpaths) == 0 {
```

Comment:
probably < 2 here as that would only be "target" mode in the list, presuming we always add prepend "target"